### DISCUSSION TITLE
Deprecation of Ethereum testnets

### DISCUSSION THREAD

**Discussion by marlowl** [Open-Attestation/open-attestation#238] [Q&A] [answered]
Ethereum is moving towards the Merge (switch from Proof-of-Work to Proof-of-Stake).  
In the meantime, a little more is known about what will happen to the testnets. (See [this](https://blog.ethereum.org/2022/06/21/testnet-deprecation/)). 

I noticed that a lot of OpenAttestation/TradeTrust tooling is still used on Rinkeby/Ropsten. As can be read in the above posted blog, Goerli will be a long term supported network. Are there any plans to migrate to or support Goerli?

**Comment by superical** [comment, marked as answer]
If I'm not wrong, it should work with most of the tooling by pointing the network to Goerli. For instance, the [TradeTrust dev site](https://dev.tradetrust.io/verify) has started supporting Goerli from its frontend several months back. If the Goerli network isn't directly available in some of these tooling, I believe passing in a provider that points to Goerli should work for most cases. Let us know if this isn't the case though.

**Reply by marlowl** [reply]
@superical The tooling is definitely build abstract enough to build customs setups indeed, for example I used this for the oa-verify package (probably what your comment is referring to):

```typescript
const providerOptions : ProviderDetails = {
  network: "goerli",
  providerType: "infura",
  url: "https://goerli.infura.io/v3/"
};

const customProvider = utils.generateProvider(providerOptions);
const verify = verificationBuilder(openAttestationVerifiers, { provider: customProvider });
```

I tried for example using the open-attestation cli for deploying token-registry and title-escrow contracts and noticed [this](https://github.com/Open-Attestation/open-attestation-cli/blob/master/src/commands/shared.ts#L55) and [this](https://github.com/Open-Attestation/open-attestation-cli/blob/master/src/implementations/config/types.ts#L25).  I build a small workaround with ethers.js and the token-registry npm package to still deploy to Goerli. I could imagine that there are more of these cases present in the current tooling (Although, as you are already referring to, workarounds/custom setups are definitely possible! :smile:).

I do see this more of 'longer' term desicions of Eth network targeting. We (Blocklab/PoR) are planning for example to target Goerli (as it will be long term supported) and probably later on Polygon(Mumbai) as networks for now.

Thanks for your response! :smile:

**Reply by superical** [reply]
@marlowl  Right, I see your point. @simboonlong is helping out at adding directly support for goerli and polygon networks within the CLI. It will be ready once the PR (https://github.com/Open-Attestation/open-attestation-cli/pull/210) gets merged.

As for the token registry package, you might want to check out the [beta](https://github.com/Open-Attestation/token-registry/tree/beta) version. This version comes with its own hh deploy scripts that work with most of the [networks](https://github.com/Open-Attestation/token-registry/tree/beta#network-configuration) you've mentioned. It's a lot more flexible since you could theoretically do whatever you want in the hh config that we haven't yet pre-configured. Here's the doc for the hh deploy commands: https://github.com/Open-Attestation/token-registry/tree/beta#deployment

Appreciate your feedback! Keep them coming, pal!😊

**Reply by superical** [reply]
@marlowl Just an update, https://github.com/Open-Attestation/open-attestation-cli/pull/210 has been merged!🎉
